adv.py

```diff
@@ -1,35 +1,126 @@
+
 from room import Room
 from player import Player
 from world import World
-
-import random
+from util import Stack, Queue  # These may come in handy
 from ast import literal_eval
+import random
+''' 
+     possible dirs --> path
+0 --> n, s, e, w --> n
+1 --> n, s --> n n
+2 --> s --> n n s
+1 --> s --> n n s s
+0 --> s e w --> n n s s s 
+5 --> n, s --> n n s s s s 
+6 --> n --> n n s s s s n
+5 --> n --> n n s s s s n n 
+0 --> e w --> n n s s s s n n w 
+7 --> e w --> n n s s s s n n w w 
+8 --> e --> n n s s s s n n w w e 
+7 --> e --> n n s s s s n n w w e e
+0 --> e --> n n s s s s n n w w e e e e
+3 --> e w --> n n s s s s n n w w e e e e e
+4 --> w --> n n s s s s n n w w e e e e e w
+3 --> w -->n n s s s s n n w w e e e e e w w
+0 --> ALL VISITED!! 
+'''
 
 # Load world
 world = World()
 
 
 # You may uncomment the smaller graphs for development and testing purposes.
 # map_file = "maps/test_line.txt"
-# map_file = "maps/test_cross.txt"
+map_file = "maps/test_cross.txt"
 # map_file = "maps/test_loop.txt"
 # map_file = "maps/test_loop_fork.txt"
-map_file = "maps/main_maze.txt"
+# map_file = "maps/main_maze.txt"
 
 # Loads the map into a dictionary
 room_graph=literal_eval(open(map_file, "r").read())
 world.load_graph(room_graph)
-
+print("the rooms", world.rooms)
 # Print an ASCII map
 world.print_rooms()
 
+# Initialize a player in the world at starting toom 0
 player = Player(world.starting_room)
 
 # Fill this out with directions to walk
 # traversal_path = ['n', 'n']
 traversal_path = []
 
-
+from util import Stack, Queue, Graph  # These may come in handy
+
+#Step 1) Instantiate Graph to traverse
+g = Graph()
+
+# #run depth first traversal to create the graph
+# g.dft(world.starting_room)
+
+opposite_directions = {
+    'n':'s',
+    's':'n',
+    'e':'w',
+    'w':'e'
+}
+
+def dft_recursive(starting_room):
+    '''
+    Should take a starting room 
+    Use a dft + a visited dict to pick random directions and travel. 
+
+    Keep dft'ing until we hit a dead end
+    Once we hit a dead end, BFS for a '?' to find a room that hasn't been visited
+
+    Repeat
+    '''
+        
+    def add_to_path_recursive(room, visited=None):
```

Comment:
I like your recursive dft implementation for this problem. Good use of docstrings to give a broad overview of the function.